eZ/Publish/API/Repository/ObjectStateService.php

```diff
@@ -66,7 +66,7 @@ public function loadObjectStateGroups($offset = 0, $limit = -1, array $prioritiz
      *
      * @return \eZ\Publish\API\Repository\Values\ObjectState\ObjectState[]
      */
-    public function loadObjectStates(ObjectStateGroup $objectStateGroup, array $prioritizedLanguages = []);
+    public function loadObjectStates(ObjectStateGroup $objectStateGroup, array $prioritizedLanguages = []): iterable;
```

Comment:
Instead of using `iterable` as a typehint, it would maybe be better to return a custom object e.g. `ObjectStateList` that implements `ArrayAccess`, `Countable` and `IteratorAggregate`, making it possible to have an object which is iterable as well as having the option of accessing the elements by their index if needed.

Reply:
Thank you for the feedback @emodric . This is out of scope right now but in the near future we would like to improve API as well :wink: